app/controllers/application_controller.rb

```diff
@@ -1018,11 +1018,28 @@ def query_params(query = nil)
   helper_method :query_params
 
   def add_query_param(params, query = nil)
-    params[:q] = get_query_param(query) unless browser.bot?
-    params
+    return params if browser.bot?
+
+    query_param = get_query_param(query)
+    if params.is_a?(String) # i.e., if params is a path
+      append_query_param_to_path(params, query_param)
+    else
+      params[:q] = query_param
+      params
+    end
```

Comment:
These changes make `add_query_param` work with Rails path helpers (in addition to controller/action hashes)